lib/house.rb

```diff
@@ -0,0 +1,47 @@
+class House
+
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price   = price
+    @address = address
+    @rooms   = []
+  end
+
+  def price
+    @price.to_s.gsub(/[$,]/,'').to_i
+  end
+
+  def add_room(name)
+    @rooms << name
+  end
+
+  def above_market_average?
+    price > 500000
+  end
+
+  def rooms_from_category(category)
+    @rooms.select do |room|
+      room.category == category
+    end
+  end
+  def total_width_of_rooms
+    @rooms.collect(&:width).map(&:to_i)
+  end
+
+  def area
+    length = @rooms.collect(&:length)
+    width = @rooms.collect(&:width).map(&:to_i)
+    combine = length.zip(width).map{|x, y| x * y}
+    combine.inject(0){|sum,x| sum + x }
+  end
+
+  def details
+    detail = {}
+    detail["price"] = price
+    # detail["address"] = @address
+  end
```

Comment:
You were really close to getting this working.  you _could_ uncomment line 44, and then return `detail` as the last line of code: 

```ruby
  def details
    detail = {}
    detail["price"] = price
    detail["address"] = @address
    detail
  end
```
OR, a simpler solution:
```ruby
  def details
    {
      "price" => price,
      "address" => @address
    }
  end
```